tests/callbacks/test_progress_bar.py

```diff
@@ -193,3 +193,43 @@ def on_test_batch_end(self, trainer, pl_module, batch, batch_idx, dataloader_idx
 
     trainer.test(model)
     assert progress_bar.test_batches_seen == progress_bar.total_test_batches
+
+
+@pytest.mark.parametrize('num_sanity_val_steps,num_val_dataloaders_batches,num_sanity_check_run_steps', [
+    (-1, [10], 10),
+    (0, [10], 0),
+    (2, [10], 2),
+    (10, [2], 2),
+    (10, [2, 3], 5),
+    (10, [20, 3], 13),
+    (10, [20, 30], 20),
+    (10, [float('inf')], 10),
+    (10, [1, float('inf')], 11),
+])
+def test_sanity_check_progress_bar_total(tmpdir, num_sanity_val_steps, num_val_dataloaders_batches, num_sanity_check_run_steps):
+    """Test that the sanity_check progress finishes with the correct total steps processed."""
+
+    tmp_model = EvalModelTemplate(batch_size=1)
+    batch_size = len(tmp_model.dataloader(train=False, num_samples=1).dataset)
+    model = EvalModelTemplate(batch_size=batch_size)
+
+    num_dataloaders = len(num_val_dataloaders_batches)
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        progress_bar_refresh_rate=1,
+        limit_val_batches=len(model.dataloader(train=False)) * num_dataloaders,
+        max_epochs=0,
+        num_sanity_val_steps=num_sanity_val_steps,
+    )
+
+    val_dataloaders = []
+    for num_samples in num_val_dataloaders_batches:
+        if num_samples == float('inf'):
+            val_dataloaders.append(model.val_dataloader__infinite())
+        else:
+            val_dataloaders.append(
+                model.dataloader(train=False, num_samples=num_samples))
+    trainer.fit(model, val_dataloaders=val_dataloaders)
+
+    val_progress_bar = trainer.progress_bar_callback.val_progress_bar
+    assert getattr(val_progress_bar, 'total', 0) == num_sanity_check_run_steps
```

Comment:
```suggestion
@pytest.mark.parametrize('num_sanity_val_steps,num_val_dataloaders_batches,expected_num_steps', [
    (-1, [10], 10),
    (0, [10], 0),
    (2, [10], 2),
    (10, [2], 2),
    (10, [2, 3], 5),
    (10, [20, 3], 13),
    (10, [20, 30], 20),
    (10, [float('inf')], 10),
    (10, [1, float('inf')], 11),
])
def test_sanity_check_progress_bar_total(tmpdir, num_sanity_val_steps, num_val_dataloaders_batches, expected_num_steps):
    """Test that the sanity_check progress finishes with the correct total steps processed."""

    tmp_model = EvalModelTemplate(batch_size=1)
    batch_size = len(tmp_model.dataloader(train=False, num_samples=1).dataset)
    model = EvalModelTemplate(batch_size=batch_size)

    num_dataloaders = len(num_val_dataloaders_batches)
    trainer = Trainer(
        default_root_dir=tmpdir,
        progress_bar_refresh_rate=1,
        limit_val_batches=len(model.dataloader(train=False)) * num_dataloaders,
        max_epochs=0,
        num_sanity_val_steps=num_sanity_val_steps,
    )

    val_dataloaders = []
    for num_samples in num_val_dataloaders_batches:
        if num_samples == float('inf'):
            val_dataloaders.append(model.val_dataloader__infinite())
        else:
            val_dataloaders.append(
                model.dataloader(train=False, num_samples=num_samples))
    trainer.fit(model, val_dataloaders=val_dataloaders)

    val_progress_bar = trainer.progress_bar_callback.val_progress_bar
    assert getattr(val_progress_bar, 'total', 0) == expected_num_steps
```

Reply:
I would always use the word "expected" for these things, so that it is immediately clear what the third argument means.

Reply:
It's a good idea!

Reply:
when the tests pass, I'll approve, I think this is ready. Good job!